app/models/node.rb

```diff
@@ -511,13 +511,36 @@ def mailing_list
 
   # used in typeahead autocomplete search results
   def icon
-    icon = 'file' if type == 'note'
-    icon = 'book' if type == 'page'
-    icon = 'map-marker' if type == 'map'
-    icon = 'flag' if has_tag('chapter')
-    icon = 'wrench' if type == 'tool'
-    icon = 'question-circle' if has_power_tag('question')
-    icon
+    # It seems the order of the conditions is important here
```

Comment:
I think using case statementa will refactor this even further...What do you think?